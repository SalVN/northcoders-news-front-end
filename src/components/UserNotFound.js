import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import './css/PageNotFound.css';

const UserNotFound = function () {
    return (
        <div className='card page-not-found-card'>
            <div className='card-image'>
                <img src='http://res.cloudinary.com/dl37xtqhv/image/upload/v1505237987/woman-565104_1920_amcmvg.jpg'/>
            </div>
            <div className='card-content'>
                <div className='page-not-found-title'>User Not Found</div>
                <div>That user does not exist</div>
                <Link to='/'><button className='button page-not-found-button'>Return to home page</button></Link>
            </div>
        </div>
    );
};

export default UserNotFound;