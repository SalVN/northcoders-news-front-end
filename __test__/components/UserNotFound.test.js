import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';


import UserNotFound from '../../src/components/UserNotFound';

describe('UserNotFound', () => {
    it('is a function', () => {
        expect(typeof UserNotFound).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<UserNotFound />);
        expect(enzymeWrapper.children().length).toEqual(2);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <UserNotFound />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains one link', () => {
        const enzymeWrapper = shallow(<UserNotFound />);
        expect(enzymeWrapper.find('Link').length).toBe(1);
    });

});