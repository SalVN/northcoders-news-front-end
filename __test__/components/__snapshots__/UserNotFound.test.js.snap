// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNotFound renders correctly 1`] = `
<div
  className="card page-not-found-card"
>
  <div
    className="card-image"
  >
    <img
      src="http://res.cloudinary.com/dl37xtqhv/image/upload/v1505237987/woman-565104_1920_amcmvg.jpg"
    />
  </div>
  <div
    className="card-content"
  >
    <div
      className="page-not-found-title"
    >
      User Not Found
    </div>
    <div>
      That user does not exist
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button page-not-found-button"
      >
        Return to home page
      </button>
    </a>
  </div>
</div>
`;
